pio-test: add page (#5815)
tldr-pages/tldr: pages/common/pio-test.md

# pio test

> Run local tests on a PlatformIO project.
> More information: <https://docs.platformio.org/en/latest/core/userguide/cmd_test.html>.

> - Run all tests in all environments of a PlatformIO project:

Reviewer: Perhaps this will read better?
```suggestion
- Run all tests in all environments of the current PlatformIO project:
```

`pio test`

- Test only specific environments:

`pio test --environment {{environment1}} --environment {{environment2}}`

> - Run only tests whose name match a specific pattern:

Reviewer: Perhaps this will read better?
```suggestion
- Run only tests whose name matches a specific pattern:
```

Also, is this a regex pattern?

> - Run only tests whose name matches a specific pattern:

Reviewer 2: ```suggestion
- Run only tests whose name matches a specific glob pattern:
```

> `pio test --filter "{{pattern}}"`

Reviewer: I think you missed my other comment - is this a regex, or something else?

Author: Something specific. It just supports a few [basic options](https://docs.platformio.org/en/latest/core/userguide/cmd_test.html#cmdoption-pio-test-f).

Reviewer 2: That'd be a glob pattern.

> - Ignore tests whose name match a specific pattern:

Reviewer: Same here
```suggestion
- Ignore tests whose name matches a specific pattern:
```

> - Ignore tests whose name matches a specific pattern:

Reviewer 2: ```suggestion
- Ignore tests whose name matches a specific glob pattern:
```

`pio test --ignore "{{pattern}}"`

- Specify a port for firmware uploading:

> `pio test --upload-port {{upload_port}}`

Reviewer: What precisely is mean  by a port here? I assume a COM port on Windows, but how does this work on Linux where we call them devices (e.g. `/dev/ttyUSB0`)?

Author: For me, it's `/dev/ttyUSB0` or `/dev/ttyACM0`, depending on the device. I just wasn't sure if this is general enough to use in in the example.

Reviewer: Ah, ok. Looks like an example of program-specific language here - so let's leave it as-is.
